src/ui/edit/main-window.c

```diff
@@ -272,6 +273,44 @@ on_song_unsaved_changed (const GObject * object, GParamSpec * arg,
   g_free (title);
   //-- release the references
   g_object_unref (song);
+  g_object_unref (song_info);
+}
+
+static void
+on_song_unsaved_changed (const GObject * object, GParamSpec * arg,
+    gpointer user_data)
+{
+  bt_main_window_update_title (BT_MAIN_WINDOW (user_data));
+}
+
+static void
+on_song_title_changed (const GObject * object, GParamSpec * arg,
+    gpointer user_data)
+{
+  bt_main_window_update_title (BT_MAIN_WINDOW (user_data));
+}
+
+static void
+on_song_changed (const GObject * object, GParamSpec * arg,
+    gpointer user_data)
+{
+  BtMainWindow *self = BT_MAIN_WINDOW (user_data);
+  bt_main_window_update_title (BT_MAIN_WINDOW (user_data));
+  
+  BtSong *song;
+
+  g_object_get (self->priv->app, "song", &song, NULL);
+  if (!song)
+    return;
+
+  BtSongInfo* song_info;
+  g_object_get (song, "song-info", &song_info, NULL);
+
+  g_signal_connect (song_info, "notify::name",
```

Comment:
I presume it's ok to do this without disconnecting the old handler because the old handler will automatically be disconnected when the old "song-info" object is finalised. Is that right?